### PR TITLE
categories-api

### PR DESCRIPTION
**Pull request by adnaani**
I will create a categories endpoint that will have an array of category objects, each of which should have the following properties, array and slug.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+process.env.NODE_ENV = "test";
```
Comment (scar1377) [inline __tests__/app.test.js]: Don't really need this line. NODE_ENV will set to "test" automatically when test is running

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+const seed = require("../db/seeds/seed");
+const testData = require("../db/data/test-data");
+const request = require("supertest");
+const app = require("../app.js");
+const db = require("../db/connection");
+// require("jest-sorted");
```
Comment (scar1377) [inline __tests__/app.test.js]: remove the comments

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            expect(category).toEqual(
+              expect.objectContaining({
+                slug: expect.any(String),
+                description: expect.any(String),
+              })
+            );
+          });
+        });
+    });
+  });
```
Comment (scar1377) [inline __tests__/app.test.js:29]: body should work. don't need _body

File: controllers/categories.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const { selectAllCategories } = require("../models/categories");
+
+exports.getAllCategories = (req, res, next) => {
+  selectAllCategories()
+    .then((categories) => {
+      res.status(200).send(categories);
```
Comment (scar1377) [inline controllers/categories.js]: sending object instead of only the array

File: listen.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const app = require("./app");
+const PORT = 8090;
+
+app.listen(PORT, (err) => {
+  if (err) {
+    console.log(err);
+  } else {
+    console.log(`Listening to ${PORT} ...`);
+  }
+});
```
Comment (scar1377) [inline listen.js]: no listen.js needed at this point

File: __tests__/app.test.js (hunk 3 above)
Comment (scar1377) [reply __tests__/app.test.js:29]: better to have an object inside body rather than only an array

**Review by scar1377 — CHANGES_REQUESTED** [review]
Please make the changes then send another /nchelp pr

File: __tests__/app.test.js [hunk 6]
```diff
@@ -15,11 +13,10 @@ describe("API: /api/categories", () => {
       return request(app)
         .get("/api/categories")
         .expect(200)
-        .then(({ _body }) => {
-          console.log(_body);
-          expect(_body).toHaveLength(4);
-          expect(_body).toBeInstanceOf(Array);
-          _body.forEach((category) => {
+        .then(({ _body: { categories } }) => {
```
Comment (popatre) [inline __tests__/app.test.js]: You dont need `_body`. The key is `body`- no underscore, its not hidden or private.

**Review by popatre — CHANGES_REQUESTED** [review]
Please change the body key to not be an underscore. 
The listen file is also not needed for this ticket - as Scarlett mentioned.

File: __tests__/app.test.js (hunk 6 above)
Comment (adnaani) [reply __tests__/app.test.js]: When i look at the body object it came up as _body, it didnt work when i did body.

**Pull request merged**